File: src/shared/config/mapConfig.ts
import Object from "@rbxts/object-utils";
import { targetConfig, trashConfig } from "./targetConfig";

export const difficulties = {
	Easy: new Color3(0, 1, 0),
	Medium: new Color3(1, 0.5, 0),
	Hard: new Color3(1, 0, 0),
};

interface MapConfig {
	targetList: Array<keyof typeof targetConfig>;
	requiredStrength: number;
	difficulty: keyof typeof difficulties;
}

export const mapConfig: Record<string, MapConfig> = {
	Grasslands: {
		targetList: [
			"Treasure Chest",
			"Ring",
			"Necklace",
			"Diamond",
			"Bag of coins",
			"Ancient Artifact",
			...Object.keys(trashConfig),
		],
		requiredStrength: 0,
		difficulty: "Easy",
	},

	Volcano: {
		targetList: [
			"Volcano rock",
			"Volcano's tear",
			"Ammonoids",
			"Heat rock",
			"Lava bucket",
			"Obsidian shard",
			"Coal artifact",
			...Object.keys(trashConfig),
		],
		requiredStrength: 100,
		difficulty: "Medium",
	},

	Frozen: {
		targetList: [
			"Ice shard",
			"Old mitten",
			"Sapphire gem",
			"Viking dagger",
			"Viking sword",
			...Object.keys(trashConfig),
		],
		requiredStrength: 500,
		difficulty: "Hard",
	},
};

File: src/shared/config/mapConfig.lua
--!optimize 2
--!native
-- Compiled with roblox-ts v3.0.0
local TS = require(game:GetService("ReplicatedStorage"):WaitForChild("rbxts_include"):WaitForChild("RuntimeLib"))
local Object =
	TS.import(script, game:GetService("ReplicatedStorage"), "rbxts_include", "node_modules", "@rbxts", "object-utils")
local trashConfig = TS.import(script, game:GetService("ReplicatedStorage"), "TS", "config", "targetConfig").trashConfig
local Sift =
	TS.import(script, game:GetService("ReplicatedStorage"), "rbxts_include", "node_modules", "@rbxts", "sift", "out")

local difficulties = {
	Easy = Color3.new(0, 1, 0),
	Medium = Color3.new(1, 0.5, 0),
	Hard = Color3.new(1, 0, 0),
}
local mapConfig = {
	Grasslands = {
		targetList = Sift.Array.concat({
			"Treasure chest",
			"Ring",
			"Necklace",
			"Diamond",
			"Bag of coins",
			"Ancient artifact",
			"Gold chalice",
			"Ruby",
			"Cool shell",
			"Golden eyepatch",
			"Jade fish",
			"Bejeweled pegleg",
		}, Object.keys(trashConfig)),
		recommendedStrength = 1,
		difficulty = "Easy",
		order = 1,
	},
	Volcano = {
		targetList = Sift.Array.concat({
			"Volcano rock",
			"Volcano's tear",
			"Ammonoids",
			"Heat rock",
			"Lava bucket",
			"Obsidian shard",
			"Coal artifact",
			"Jade statue",
			"T-Rex tooth",
			"T-Rex skull",
			"Skull",
			"Obsidian crown",
		}, Object.keys(trashConfig)),
		recommendedStrength = 12,
		difficulty = "Easy",
		order = 2,
	},
	Frozen = {
		targetList = Sift.Array.concat({
			"Frozen ring",
			"Frozen teddy",
			"Icecicle",
			"Sapphire gem",
			"Viking dagger",
			"Viking sword",
			"Topaz amulet",
			"Gift",
			"Small gift",
			"Mammoth tusk",
			"Diamond yeti",
			"Viking helmet",
		}, Object.keys(trashConfig)),
		recommendedStrength = 50,
		difficulty = "Easy",
		order = 3,
	},
}
return {
	difficulties = difficulties,
	mapConfig = mapConfig,
}
